tool-plugins/vscode/src/extension.ts

```diff
@@ -91,6 +91,21 @@ export function activate(context: ExtensionContext): Promise<any> {
         activateTreeView(ballerinaExtInstance);
         // Enable Ballerina Syntax Highlighter
         activateSyntaxHighlighter(ballerinaExtInstance);
+
+        ballerinaExtInstance.onReady().then(() => {
+            const langClient = <ExtendedLangClient>ballerinaExtInstance.langClient;
+            // Send initial configuration 
+            const args: DidChangeConfigurationParams = {
+                settings: workspace.getConfiguration('ballerina'),
+            };
+            langClient.sendNotification("workspace/didChangeConfiguration", args);
```

Comment:
I think this is to capture the initial configuration at the server-side? It seems that we are violating the LSP by sending the didChangeConfiguration notification through the extension bypassing the protocol. I think what we should do to fetch the initial configuration is to use the configuration request at the server-side [1].
[1] https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#workspace_configuration

Reply:
+1, initially I was reluctant to add this `workspace/configuration` request since some clients might not have implemented it. If we stressing down on the protocol let's strict with it, I'll add a fix.

Reply:
> +1, initially I was reluctant to add this `workspace/configuration` request since some clients might not have implemented it. If we stressing down on the protocol let's strict with it, I'll add a fix.

Awesome :D

Reply:
As discussed since we are having troubles in getting the result of the CompletableFuture, will do this as an improvement. Created https://github.com/ballerina-platform/ballerina-lang/issues/21390 to track this.